sdk/python/feast/data_source.py

```diff
@@ -220,20 +220,42 @@ def to_proto(self) -> DataSourceProto.KinesisOptions:
 
 class DataSource(ABC):
     """
-    DataSource that can be used source features
+    DataSource that can be used to source features.
+
+    Args:
+        event_timestamp_column (optional): Event timestamp column used for point in time
+            joins of feature values.
+        created_timestamp_column (optional): Timestamp column indicating when the row
+            was created, used for deduplicating rows.
+        field_mapping (optional): A dictionary mapping of column names in this data
+            source to feature names in a feature table or view. Only used for feature
+            columns, not entity or timestamp columns.
+        date_partition_column (optional): Timestamp column used for partitioning.
     """
 
+    _event_timestamp_column: str
+    _created_timestamp_column: str
+    _field_mapping: Dict[str, str]
+    _date_partition_column: str
+
     def __init__(
         self,
-        event_timestamp_column: Optional[str] = "",
-        created_timestamp_column: Optional[str] = "",
+        event_timestamp_column: Optional[str] = None,
```

Comment:
I'm curious about these type annotation changes. Are they necessary for the docs?

Reply:
These changes are not strictly necessary to enforce proper typing. However, with the default value set to `None`, I believe it's easier for a user to see that the line `self._event_timestamp_column = event_timestamp_column if event_timestamp_column else ""` will leave an input string untouched, and otherwise will set `self._event_timestamp_column` to the empty string. In general, if there is a parameter of type `Optional[X]` and a `None` value will lead to some inference, I think it's more clear to set the default value to `None`.